backend/src/app/middleware/userJwtTokenDecoder.ts

```diff
@@ -15,8 +15,7 @@ export class UserJWTTokenDecoderMiddleware implements NestMiddleware {
 
   async use(req: Request, res: Response, next: NextFunction) {
     const token = req.headers['authorization']?.split(' ')[1];
-
-    if (token) {
+    if (token && token !== 'undefined' && token.trim() !== '' ) {
```

Comment:
token?.trim() can do the same check

Reply:
Here we receive token as 'undefined' string when user not logged in